src/aggregation/mean.py

```diff
@@ -5,9 +5,9 @@
 """
 
 from torch import nn
-
+import torch.nn.functional as F
 class Mean(nn.Module):
   def __init__(self):
     super().__init__()
   def forward(self, x):
-    return x.mean(axis=-2) # assumes shape always ends with (..., n_frames, embed_dim)
+    return F.normalize(x.mean(axis=-2), dim=-1) # assumes shape always ends with (..., n_frames, embed_dim)
```

Comment:
I think let's keep it as is right now and do a large 'normalizing' themed refactoring next, it's good you bring it up.